File: dynamic_listings/C++/default/collisiondetection/point_circle.cpp
#include <cmath>

struct Circle{
    Point center;
    float radius;
};

float distance(Point A, Point B){
    // Calculates the distance between two points
    return std::sqrt(std::pow((A.x - B.x),2) + std::pow((A.y - B.y),2));
}

bool circle_point_collision(Circle A, Point B){
    return distance(A.center, B) <= A.radius;
}

File: dynamic_listings/rust/default/collisiondetection/point_circle.rs
# #![allow(dead_code)]
# struct Point {
#     x: f32,
#     y: f32,
# }
struct Circle {
    center: Point,
    radius: f32,
}

fn distance(a: &Point, b: &Point) -> f32 {
    // Calculates the distance between two points
    ((a.x - b.x).powi(2) as f32 + (a.y - b.y).powi(2) as f32).sqrt()
}

fn circle_point_collision(a: &Circle, b: &Point) -> bool {
    distance(&a.center, b) <= a.radius
}
